integration-tests/src/test/java/com/datastax/oss/driver/core/resolver/MockResolverIT.java

```diff
@@ -59,47 +58,148 @@ public class MockResolverIT {
+        filteredNodes =
+            nodes.stream()
+                .filter(x -> x.toString().contains("node-1.cluster.fake"))
+                .collect(Collectors.toSet());
+        assertThat(filteredNodes).hasSize(1);
+        InetSocketAddress address =
+            (InetSocketAddress) filteredNodes.iterator().next().getEndPoint().resolve();
+        assertTrue(address.isUnresolved());
+      }
+    }
+  }
 
-    MockResolverFactory resolverFactory = new MockResolverFactory();
-    resolverFactory.updateResponse(
-        "node-1.cluster.fake",
-        new ValidResponse(new InetAddress[] {getNodeInetAddress(ccmBridge, 1)}));
-    ResolverProvider.setDefaultResolverFactory(resolverFactory);
+  public void replace_cluster_test() {
 
     DriverConfigLoader loader =
         new DefaultProgrammaticDriverConfigLoaderBuilder()
             .withBoolean(TypedDriverOption.RESOLVE_CONTACT_POINTS.getRawOption(), false)
             .withBoolean(TypedDriverOption.RECONNECT_ON_INIT.getRawOption(), true)
             .withStringList(
                 TypedDriverOption.CONTACT_POINTS.getRawOption(),
-                Collections.singletonList("node-1.cluster.fake:9042"))
+                Collections.singletonList("test.cluster.fake:9042"))
             .build();
 
     CqlSessionBuilder builder = new CqlSessionBuilder().withConfigLoader(loader);
-    try (CqlSession session = builder.build()) {
+    CqlSession session;
+
+    try (CcmBridge ccmBridge = CcmBridge.builder().withNodes(3).withIpPrefix("127.0.1.").build()) {
+      MockResolverFactory mockResolverFactory = new MockResolverFactory();
+      mockResolverFactory.updateResponse(
+          "test.cluster.fake",
+          new ValidResponse(
+              new InetAddress[] {
+                getNodeInetAddress(ccmBridge, 1),
+                getNodeInetAddress(ccmBridge, 2),
+                getNodeInetAddress(ccmBridge, 3)
+              }));
+      ResolverProvider.setDefaultResolverFactory(mockResolverFactory);
+      ccmBridge.create();
+      ccmBridge.start();
+      session = builder.build();
+      for (int i = 0; i < 30; i++) {
+        try {
+          Collection<Node> nodes = session.getMetadata().getNodes().values();
+          if (nodes.size() == 3) {
+            break;
+          }
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
+          break;
+        }
+      }
+      ResultSet rs = session.execute("SELECT * FROM system.local");
+      assertThat(rs).isNotNull();
+      Row row = rs.one();
+      assertThat(row).isNotNull();
+      Collection<Node> nodes = session.getMetadata().getNodes().values();
+      assertThat(nodes).hasSize(3);
+      Iterator<Node> iterator = nodes.iterator();
+      while (iterator.hasNext()) {
+        LOG.trace("Metadata node: " + iterator.next().toString());
+      }
+      Set<Node> filteredNodes;
+      filteredNodes =
+          nodes.stream()
+              .filter(x -> x.toString().contains("test.cluster.fake"))
```

Comment:
If it fails here, could you please add another part of the test when cluster is killed and recreated with another IPs, which are getting registered right away in `MockResolverFactory`.
And make sure that session become operation in couple of seconds after it happens.

Probably we stumbled at https://github.com/scylladb/scylla-enterprise/issues/4230 repro

Reply:
This is section where we still interact with the first cluster - the test does not fail here. Do you mean that I should change the prefix of the second cluster to something different?

Reply:
you need to spot it, throw a log message it happened but don't fail the test.

Reply:
done